packages/core/upload/admin/src/components/BulkMoveDialog/BulkMoveDialog.js

```diff
@@ -13,21 +14,38 @@ import {
 import { FieldLabel } from '@strapi/design-system/Field';
 import { Stack } from '@strapi/design-system/Stack';
 import { Typography } from '@strapi/design-system/Typography';
-import { Form } from '@strapi/helper-plugin';
+import { Form, getAPIInnerErrors } from '@strapi/helper-plugin';
 
+import { useBulkMove } from '../../hooks/useBulkMove';
 import { getTrad } from '../../utils';
 import SelectTree from '../SelectTree';
 import { useFolderStructure } from '../../hooks/useFolderStructure';
+import { FolderDefinition, AssetDefinition } from '../../constants';
 
-export const BulkMoveDialog = ({ onClose, errors: initialErrors }) => {
+export const BulkMoveDialog = ({ onClose, selected }) => {
   const { formatMessage } = useIntl();
   const { data: folderStructure, isLoading } = useFolderStructure();
+  const { move } = useBulkMove();
   const initialFormData = {
     destination: folderStructure[0],
   };
 
-  const handleSubmit = values => {
-    onClose({ moved: true, destinationFolderId: values.destination.value });
+  const handleSubmit = async (values, { setErrors }) => {
+    try {
+      await move(values.destination.value, selected);
+      onClose();
+    } catch (error) {
+      const errors = getAPIInnerErrors(error, { getTrad });
+      const formikErrors = Object.entries(errors).reduce((acc, [key, error]) => {
+        acc[key || 'destination'] = error.defaultMessage;
+
+        return acc;
+      }, {});
+
+      if (!isEmpty(formikErrors)) {
+        setErrors(formikErrors);
```

Comment:
I'm wondering how we should handle this copy, I think the way we handle it now doesn't allow traduction (because not linked to a trad key). Not sure I remember how we handle those types of backend error in the admin to make sure we get a trad key for it 🤔 WDYT?

Reply:
In theory it is possible to be translated, but at the moment we do not apply it anywhere. I'd like to work on this soon, but this requires the error refactoring Vicent and I started a while ago.

Reply:
thanks for the information, I remember it now, will be nice indeed!